Core/Rewriters/Formatter.cs

```diff
@@ -143,11 +143,28 @@ private static InvocationExpressionSyntax FormatInvocationExpression (Invocation
 
     private static MemberAccessExpressionSyntax FormatMemberAccessExpression (MemberAccessExpressionSyntax node)
     {
-      var indentation = ((MemberAccessExpressionSyntax) node.DescendantNodes().First (s => s.IsKind (SyntaxKind.SimpleMemberAccessExpression))).OperatorToken.LeadingTrivia.ToString();
+      var indentation = GetIndentation (node);
       return MoqSyntaxFactory.MemberAccessExpression (
           node.Expression.WithTrailingTrivia (SyntaxFactory.Whitespace (Environment.NewLine + indentation)),
           node.Name);
     }
+    private static string GetIndentation (MemberAccessExpressionSyntax node)
+    {
+      var firstMemberAccessExpressionTrivia = ((MemberAccessExpressionSyntax) node.GetFirstIdentifierName().Parent!).OperatorToken.LeadingTrivia.ToFullString();
+      if (!string.IsNullOrEmpty (firstMemberAccessExpressionTrivia))
+      {
+        return firstMemberAccessExpressionTrivia;
+      }
+
+      var secondMemberAccessExpressionTrivia = (node.GetFirstIdentifierName().Ancestors().Where (s => s.IsKind (SyntaxKind.SimpleMemberAccessExpression))
+          .Skip (1).FirstOrDefault() as MemberAccessExpressionSyntax)?.OperatorToken.LeadingTrivia.ToFullString();
+      if (!string.IsNullOrEmpty (secondMemberAccessExpressionTrivia))
+      {
+        return secondMemberAccessExpressionTrivia;
+      }
+
+      return string.Empty;
+    }
```

Comment:
Wouldn't this be very limited? What about member access expressions of the form `A.B.C.D.E`?

Reply:
i don't have to check more `MemberAccessExpressions` for their indentation, because in most cases the indentation of one of the 2 last `MemberAccessExpressions`  fits